internal/service/sso/resource_identity_provider.go

```diff
@@ -453,10 +454,44 @@ func (r *IdentityProviderResource) Schema(ctx context.Context, req resource.Sche
 				providerAttributeList,
 			),
 
-			"linkedin": identityProviderSchemaAttribute(
+			"linkedin": schema.SingleNestedAttribute{
```

Comment:
Assuming the DeprecationMessage is the only real change here, you could get the output of the identityProviderSchemaAttribute method higher up and then tack on the DeprecationMethod, so that you don't have to rewrite the rest of the single nested attr schema.